src/commands/defi/watchlist/add-nft.ts

```diff
@@ -0,0 +1,121 @@
+import { Command } from "types/common"
+import { MessageSelectOptionData, SelectMenuInteraction } from "discord.js"
+import { defaultEmojis, thumbnails } from "utils/common"
+import {
+  composeDiscordSelectionRow,
+  getSuccessEmbed,
+  composeDiscordExitButton,
+  composeEmbedMessage,
+} from "utils/discordEmbed"
+import { CommandChoiceHandler } from "utils/CommandChoiceManager"
+import { PREFIX } from "utils/constants"
+import defi from "adapters/defi"
+import { getCommandArguments } from "utils/commands"
+import CacheManager from "utils/CacheManager"
+import { handleUpdateWlError } from "../watchlist_slash"
+import { ResponseCollectionSuggestions } from "types/api"
+
+const handler: CommandChoiceHandler = async (msgOrInteraction) => {
+  const interaction = msgOrInteraction as SelectMenuInteraction
+  const value = interaction.values[0]
+  const [symbol, address, chain, userId] = value.split("_")
+  const { ok, error } = await defi.addNFTToWatchlist({
+    user_id: userId,
+    collection_symbol: symbol,
+    collection_address: address,
+    chain,
+  })
+  if (!ok) handleUpdateWlError(msgOrInteraction, symbol, error)
+  CacheManager.findAndRemove("watchlist", `watchlist-nft-${userId}`)
+  return {
+    messageOptions: {
+      embeds: [
+        getSuccessEmbed({
+          title: "Successfully set!",
+          description: `${symbol} has been added successfully! To see your watchlist use \`$wl view\``,
+        }),
+      ],
+      components: [],
+    },
+  }
+}
+
+const command: Command = {
+  id: "watchlist_nft_add",
+  command: "add-nft",
+  brief: "Add a NFT to your watchlist.",
+  category: "Defi",
+  run: async (msg) => {
+    const symbol = getCommandArguments(msg)[2]
+    const userId = msg.author.id
+    const { data, ok, error } = await defi.addNFTToWatchlist({
+      user_id: userId,
+      collection_symbol: symbol,
+    })
+    if (!ok) handleUpdateWlError(msg, symbol, error)
+    // no data === add successfully
+    if (!data) {
+      CacheManager.findAndRemove("watchlist", `watchlist-nft-${userId}`)
+      return {
+        messageOptions: {
+          embeds: [
+            getSuccessEmbed({
+              title: "Successfully set!",
+              description: `${symbol} has been added successfully! To see your watchlist use \`$wl view\``,
+            }),
+          ],
+        },
+      }
+    }
+
+    // allow selection
+    const { suggestions = [] } = data
+    const opt = (
+      collection: ResponseCollectionSuggestions
+    ): MessageSelectOptionData => ({
+      label: `${collection.name} (${collection.symbol})`,
+      value: `${collection.symbol}_${collection.address}_${collection.chain}_${msg.author.id}`,
+    })
+    const selectRow = composeDiscordSelectionRow({
+      customId: "watchlist_nft_selection",
+      placeholder: "Make a selection",
+      options: suggestions.map(opt),
```

Comment:
`suggestions` can be empty which may cause error

Reply:
I did handle the empty case